### DISCUSSION TITLE
Completion categories

### DISCUSSION THREAD

**Discussion by k4leg** [bbatsov/projectile#1787] [Ideas]
Emacs has different completion styles.  It would be nice if Projectile had categories like `projectile-find-file` or `projectile-find-dir`, or at least used built-in categories (like use `file` category for `projectile-find-file` command).

**Comment by bbatsov** [comment]
I'm not sure what those categories you refer to are, so some examples would be useful.

**Reply by k4leg** [reply]
For example, the following code adds a useful `partial-completion` completion style for `file` category.  Thus, a completion like `~/.l/s/fo` can expand to `~/.local/share/fonts`, and can be used in the `find-file` (or similar) command.

```emacs-lisp
(setq completion-styles '(basic orderless))
(setq completion-category-overrides
      '((file (styles . (basic partial-completion orderless)))))

```
